src/irmin-pack/unix/chunked_suffix_intf.ml

```diff
@@ -65,6 +75,15 @@ module type S = sig
     chunk_num:int ->
     (t, [> open_error ]) result
 
+  val new_chunk :
+    root:string ->
+    end_poff:int63 ->
+    dead_header_size:int ->
+    auto_flush_threshold:int ->
+    auto_flush_procedure:Ao.auto_flush_procedure ->
+    t ->
+    (unit, [> add_new_error ]) result
```

Comment:
A couple of suggestions to reinforce this is modifying `t`:

1. Remove `root`, `end_poff` and (possibly) `dead_header_size`. 
    - This would require storing `root` on `t` which I think is okay. In the future, perhaps we pass a path or use a chunk idx -> path function that is given at open time ([context](https://github.com/mirage/irmin/pull/2115#discussion_r995999203)).
    - `end_poff` is not strictly needed for this operation since the read only chunk is opend as not appendable and thus reads the size of the path. ([I hope we can clean this up at some point](https://github.com/mirage/irmin/pull/2115#discussion_r995922877)).
    - `dead_header_size` is necessarily 0 for this but maybe we want to still pass it. I'm on the fence. I'm sort of unsure how to handle this weird legacy thing.
2. Rename to `add_chunk` since it is modifying `t` and not returning a chunk (which it should not).

Reply:
Applied your suggestions, it is an improvement, thanks.

I'm not sure about the `end_poff`, which I thing I prefer that we read from the control file instead of using `Io.size_of_path`.